ada_feeding/ada_feeding/behaviors/moveit2/servo_move.py

```diff
@@ -118,6 +117,26 @@ def update(self) -> py_trees.common.Status:
         twist.header.stamp = self.node.get_clock().now().to_msg()
         self.pub.publish(twist)
 
+        # Write the remaining distance
+        duration = self.blackboard_get("duration")
+        self.curr_distance = self.blackboard_get("curr_distance")
+        if self.curr_distance is None:
+            self.curr_distance = duration_to_float(duration) - duration_to_float(
```

Comment:
@amalnanavati Bug

You need to check whether blackboard_get("duration") is a Duration or a float. If the latter, this throws an error.